stripe/src/main/java/com/stripe/android/paymentsheet/PaymentOptionsActivity.kt

```diff
@@ -116,12 +113,13 @@ internal class PaymentOptionsActivity : BaseSheetActivity<PaymentOptionResult>()
                 PaymentOptionResult.Failed(it)
             )
         }
-        bottomSheetController.shouldFinish.observe(this) { shouldFinish ->
-            if (shouldFinish) {
-                finish()
-            }
-        }
-        bottomSheetController.setup()
+
+        // Fragment is added with animation immediately after activity is created, so the sheet can
+        // be expanded now and it'll animate from the bottom.
+        bottomSheetController.expand()
+        // isFitToContents causes conflicts when calculating the sheet position upon resize.
+        // CoordinatorLayout will already position the sheet correctly with gravity = bottom.
+        bottomSheetBehavior.isFitToContents = false
```

Comment:
If the layout changes while the sheet is transitioning from `STATE_HIDDEN` to `STATE_SETTLING` to `STATE_EXPANDED` it can cause layout measurement issues, which result in the sheet being positioned incorrectly. If the sheet is expanded from the beginning, like here, it will not animate in from the bottom. That's fine here because when adding the fragment the layout will animate to accommodate it. PaymentSheet has a fragment from start, so we need to start as hidden otherwise the sheet will not animate from the bottom, and will just fade in with the rest of the activity. I'll try to come up with a more structured solution that encodes this knowledge.

Reply:
Updated the PR description. Still not perfect but changed a little the implementation to get the two use cases closer together.